### DISCUSSION TITLE
What does scoping to request instead of thread mean exactly in versions 3+?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [pallets-eco/flask-sqlalchemy#1376] [Q&A] [answered]
Hey. The current pattern we use in v2 for managing sessions is to create a single `SQLAlchemy` object and use it to initialize sessions when needed within the code. Should this change with v3 or the scoping to request is done under the hood?
Here are possibly related questions: https://github.com/pallets-eco/flask-sqlalchemy/discussions/1374, https://github.com/pallets-eco/flask-sqlalchemy/discussions/1375

**Comment by davidism** [comment, marked as answer]
Request contexts (and app contexts) are essentially scoped the same, because each request is handled by one thread. Nothing should have changed for you.

That said, from your questions it sounds like you're doing a lot of plain-SQLAlchemy stuff rather than using Flask-SQLAlchemy typical patterns. You'd probably be better served by Flask-SQLAlchemy-Lite: https://flask-sqlalchemy-lite.readthedocs.io/en/latest/
